python_modules/dagster/dagster/_core/definitions/asset_spec.py

```diff
@@ -16,29 +16,40 @@
 if TYPE_CHECKING:
     from dagster._core.definitions.asset_dep import AssetDep, CoercibleToAssetDep
 
-# SYSTEM_METADATA_KEY_ASSET_EXECUTION_TYPE lives on the metadata of an asset
+# This lives on the metadata of an asset
 # (which currently ends up on the Output associated with the asset key)
-# whih encodes the execution type the of asset. "Unexecutable" assets are assets
-# that cannot be materialized in Dagster, but can have events in the event
-# log keyed off of them, making Dagster usable as a observability and lineage tool
-# for externally materialized assets.
-SYSTEM_METADATA_KEY_ASSET_EXECUTION_TYPE = "dagster/asset_execution_type"
+# and encodes the capability type the of backing AssetsDefinition.
+# Assets backed by AssetsDefinitions with no capabilities cannot be materialized in Dagster,
+# but can have events in the event log keyed off of them, making Dagster usable as
+# an observability and lineage tool for externally materialized assets.
+SYSTEM_METADATA_KEY_ASSET_DEF_CAPABILITY = "dagster/asset_def_capability"
 
 
-class AssetExecutionType(Enum):
-    UNEXECUTABLE = "UNEXECUTABLE"
-    MATERIALIZATION = "MATERIALIZATION"
+class AssetsDefCapability(Enum):
```

Comment:
not sure this is the right end state, but thought this was an interesting lens to try out - what the capability of the backing assets definition is